test/index.test.ts

```diff
@@ -42,6 +43,61 @@ describe('FiatConnect SDK', () => {
     expect(client.config.providerName).toEqual(exampleProviderName)
     expect(client.config.iconUrl).toEqual(exampleIconUrl)
   })
+  describe('getClock', () => {
+    it('gets the server clock', async () => {
+      fetchMock.mockResponseOnce(JSON.stringify(mockClockResponse))
+      const response = await client.getClock()
+      expect(fetchMock).toHaveBeenCalledWith(
+        'https://fiat-connect-api.com/clock',
+        expect.objectContaining({
+          method: 'GET',
+        }),
+      )
+      expect(response.ok).toBeTruthy()
+      expect(response.val).toMatchObject(mockClockResponse)
+    })
+    it('handles fetch errors', async () => {
+      fetchMock.mockRejectOnce(new Error('fake error message'))
+      const response = await client.getClock()
+
+      expect(response.ok).toBeFalsy()
+      expect(response.val).toMatchObject({
+        error: 'fake error message',
+      })
+    })
+  })
+  describe('getClockDiff', () => {
+    it('calculates clock difference when server is ahead', async () => {
+      jest
+        .spyOn(global.Date, 'now')
+        .mockReturnValueOnce(new Date('2022-05-02T22:05:55+0000').getTime())
+        .mockReturnValueOnce(new Date('2022-05-02T22:05:56+0000').getTime())
```

Comment:
this appears to rely on the implementation details of the method, rather than its high level behavior. Seems like it could be cleaner to put the diff algorithm in its own helper that takes t0, t1, t2, and t3 as parameters, and run tests on that instead. The test might also be easier to read.

Reply:
Yep, agreed; pulled this out into its own helper.